customize_image.sh

```diff
@@ -85,7 +83,10 @@ cleanup() {
 	cleanup_losetup
 }
 
+# Enable the Exist when non-zero return get catched
+set -ex
```

Comment:
I would if anything move it to the top of the file. Then it is clear that this is globally set. As this option is not only relevant for the trap.

Reply:
with -x the output message (usage() or error ) will be contaminated by the executed commands
 _-x  Print commands and their arguments as they are executed._

Reply:
the -e cause exist when command ends with non-zero, which is also not wished.
_-e  Exit immediately if a command exits with a non-zero status._

Reply:
exist != exit
Yes -x is completely independent from the trap. So putting this set command above the trap is misleading. If you want to move it, move it to the top of the file. This is the place where this is expected.

Reply:
thanks for the catch of "exist => exit". 
And for set -e, would take another example of following code, if the set -x is used before it, the "if ... " will not be executed, but I wish it get executed to give a usage message.
```
# get the options
MYOPTS=`getopt -o mh --long minimize,help -- "$@"`
if [ "$?" != 0 ]; then  usage; exit 1; fi
```

Reply:
And for the set -x, with it on the top, the output of 
`~/revpi/imagebakery$ ./customize_image.sh --noexistoptions
getopt: unrecognized option '--noexistoptions'
`
would be 
`~/revpi/imagebakery$ ./customize_image.sh --noexistoptions
++ which curl
+ '[' '!' -x /usr/bin/curl ']'
++ which fsck.vfat
+ '[' '!' -x /usr/sbin/fsck.vfat ']'
++ which lsof
+ '[' '!' -x /usr/bin/lsof ']'
++ which parted
+ PARTED=/usr/sbin/parted
+ '[' x/usr/sbin/parted = x ']'
+ '[' '!' -x /usr/sbin/parted ']'
+ '[' 1036 '!=' 2 ']'
+ '[' -x /usr/bin/newpid ']'
+ MINIMG=0
++ getopt -o mh --long minimize,help -- --noexistoptions
getopt: unrecognized option '--noexistoptions'
+ MYOPTS=' --'
+ '[' 1 '!=' 0 ']'
+ usage
+ echo 'usage: customize_image.sh [-m, --minimize | -h, --help] <image>
  -m, --minimize        build the minimized image
  -h, --help            print the usage page
If no option is specified the normal image will be built.'
usage: customize_image.sh [-m, --minimize | -h, --help] <image>
  -m, --minimize        build the minimized image
  -h, --help            print the usage page
If no option is specified the normal image will be built.
+ exit 1
`

Reply:
Personally I use -x only for debugging purpose, not for script that will be delivered.

Reply:
good idea, I will try it without -x

Reply:
Please leave the -x option for now. It is not part of your ticket to change the overall output of the tool. We have really bad error handling in the imagebakery. Removing the -x option will leave us with no clue what went wrong if a customer reports an error.

Reply:
No, not now. Your ticket is clear, please do what the ticket describes.